_posts/2023-10-23-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,311 @@
+The operator communicates with the Kafka cluster and ensures that the specified Kafka topic is created or updated according to the desired state.
+It can be deployed standalone, or along with your Kafka cluster using the `Kafka` custom resource.
+
+The first generation of the Topic Operator is based on the Kafka Streams library, and uses ZooKeeper to gather topic metadata.
+We call it **Bidirectional Topic Operator (BTO)**, as it allows to reconcile changes coming from both Kubernetes and Kafka using a three-way merge logic.
+In order to do this, it has to keep an internal topic store, which is the source-of-truth used to determine the necessary changes for synchronization.
+If the Kafka-side configuration is changed out-of-band (e.g. by using the Admin client, or Kafka shell scripts), those changes will eventually be reverted.
+
+The new generation of the Topic Operator is called **Unidirectional Topic Operator (UTO)**.
+The UTO improves on scalability and maintenance, but also provides support for Kafka clusters running in KRaft mode.
+It is fully backwards compatible with the BTO, but it will only reconcile topic events in one direction, from Kubernetes to Kafka.
+Your `KafkaTopic` custom resources will be the source-of-truth, like with all the other operators.
+
+The operator detects cases where multiple `KafkaTopic` resources are attempting to manage a Kafka topic using the same `.spec.topicName`.
+Only the oldest resource will be reconciled, while the other resource will fail with a `ResourceConflict` error.
+
+We still do not support decreasing `.spec.partitions` and changing `.spec.replicas`.
+While the former is not supported by Kafka itself, support for the latter could be added in the future.
+
+In addition to be able to pause reconciliations, the operator now also provides a way to unmanage a Kafka topic using annotations.
+If the `KafkaTopic` is paused and we delete it, then it will be also deleted in Kafka.
+If the `KafkaTopic` is unmanaged and we delete it, then only the resource will be garbage collected by Kubernetes, but the topic will still exist in Kafka.
+
+In terms of requirements, the UTO assumes the following access rights:
+
+- describe all topics
+- describe all topics configs
+- create topics
+- create partitions
+- delete topics
+- list partition reassignments
+- describe broker configs
+
+#### Scalability
+
+The BTO is limited in terms of scalability, as it only operates on one topic at a time, and it has to update a persistent store containing topic metadata.
+Instead, the UTO does not store topic metadata and it aims to be scalable in terms of the number of topics that it can operate on.
+
+When running Kafka operations, the UTO makes use of the request batching supported by the Kafka Admin client to get higher throughput for metadata operations.
+All `KafkaTopic` events are queued when received, and then processed in batches by a number of controller threads (currently, only a single thread is supported).
```

Comment:
What's a `KafkaTopic` event? Could you clarify please? I could imagine it's about the creation/update/deletion of the custom resource, or? Can you clarify it?